feat: add a global nutrient calculator that calculates nutrition for a list of products

Sunhao86 opened the pull request:
### What
This PR adds localStorage support to the nutrition calculator.

### Screenshot
![截图 2025-03-22 15-23-20](https://github.com/user-attachments/assets/0179b17b-1ca4-4288-910c-3ee170f05d91)
![截图 2025-03-22 15-23-46](https://github.com/user-attachments/assets/d4d77d27-bf78-4c19-b47f-0e1998846f8b)


### Features
Save calculator items to localStorage when updated
- Add calculator button to navigation bar
- Add "Add to Calculator" button on product pages

### Benefits
Users can now keep their calculator data between page refreshes and browser sessions.

VaiTon commented:
1. Please run `pnpm format` ALWAYS before committing
2. You can see from your own images that the OFF logo gets too little. We should move the "calculator" button to another place

teolemon reviewed (dismissed):
- This Calculator only makes sense on the product page. It should be hidden elsewhere
- Calculator should be capitalized
- We could put it in a floating and customizable action bar similar to the Mobile app. WDYT @VaiTon ?

VaiTon commented:
> We could put it in a floating and customizable action bar similar to the Mobile app. WDYT @VaiTon ?

Good with me, although I would not know where to put it xD

Sunhao86 commented:
Hi @VaiTon , I've updated the PR with your feedback:

- Removed the Calculator button from the home page .
- The Calculator is now only available on the product detail page.
- I've moved the Calculator into a floating button located at the bottom-right corner of the screen, similar to the style in the mobile app.
- I’ve run `pnpm format` to ensure all modified files follow the code formatting guidelines.

Could you please help me verify if the formatting is now correct? 🙏  Also, I would really appreciate any further suggestions you might have — I’m very happy to keep improving this! Thank you so much for your time and guidance!

![截图 2025-03-23 10-10-17](https://github.com/user-attachments/assets/0bfca252-d807-4d62-bd96-51dd7d3b95b5)
![截图 2025-03-23 10-13-55](https://github.com/user-attachments/assets/a9cb95e0-b71b-4307-b5e1-440bd363cd38)

VaiTon commented on src/lib/stores/calculatorStore.ts:
```diff
@@ -0,0 +1,99 @@
+	};
+};
+
+function loadFromStorage(): CalculatorItem[] {
```
Please use `svelte-local-storage-store`

VaiTon commented on src/lib/stores/calculatorStore.ts:
```diff
@@ -0,0 +1,99 @@
+		const existingIndex = items.findIndex((i) => i.id === item.id);
+		if (existingIndex >= 0) {
+			const updatedItems = [...items];
+			updatedItems[existingIndex].quantity += 100;
```
What is this constant `100` ?

VaiTon commented on src/lib/stores/calculatorStore.ts:
```diff
@@ -0,0 +1,99 @@
+	isCalculatorOpen.update((value) => !value);
+}
+
+export function extractNutriments(nutriments: Nutriments): CalculatorItem['nutriments'] {
```
Extract `CalculatorItem['nutriments']` to a type and use it

VaiTon commented on src/lib/ui/NutritionCalculator.svelte:
```diff
@@ -0,0 +1,152 @@
+			</div>
+
+			<!-- Total nutrition info -->
+			{@const totals = calculateTotals()}
```
Make this a `$derived` of something

VaiTon reviewed (changes requested):
Generally looks good. Just a few comments

Sunhao86 commented:
Hi @VaiTon , I've made the changes as requested, when you get a chance, I’d be grateful if you could take another look. Please let me know if there’s anything else I can improve :pray:  :pray:

VaiTon reviewed (approved):
LGTM

VaiTon commented:
@Sunhao86 please fix lint issues: https://github.com/openfoodfacts/openfoodfacts-explorer/actions/runs/14183737364/job/39735461380?pr=336

Sunhao86 commented:
Hi @VaiTon , I believe all issues are now fixed. Please let me know if there's anything else. Thank you so much🙏🙏🙏

VaiTon commented:
@Sunhao86 sorry but we've merged a quite big PR regarding translations. If you please rebase  then we're finally ok to merge!

Sunhao86 commented:
hi @VaiTon ,I've successfully rebased the calculator-storage branch onto the latest main.

No worries at all about the changes — I'm really glad to see the project progressing and growing in other areas too. Working together and improving together is exactly what makes this project so great!

abrahamvg commented:
@Sunhao86 @VaiTon 
![image](https://github.com/user-attachments/assets/ed327c62-2a5f-421e-85b1-1fb97da91c22)
